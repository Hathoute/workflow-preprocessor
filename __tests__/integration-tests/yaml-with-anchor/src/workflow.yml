name: Workflow
on:
  push:
    branches:
      - main
  pull_request:
    branches:
      - main

imports:
  - templates/template.yml

internals:
  test-template: &test_template
    extends: template/test-base
  deploy-template: &deploy_template
    extends: template/deploy-base

jobs:
  test-macos:
    <<: *test_template
    runs-on: macos-latest
    name: Test on macos
  deploy-macos:
    <<: *deploy_template
    runs-on: macos-latest
    name: Deploy on macos
    needs: [test-macos]

  test-ubuntu:
    runs-on: ubuntu-latest
    name: Test on ubuntu
    <<: *test_template
  deploy-ubuntu:
    runs-on: ubuntu-latest
    name: Deploy on ubuntu
    needs: [test-ubuntu]
    <<: *deploy_template

  test-windows:
    runs-on: windows-latest
    name: Test on windows
    <<: *test_template
  deploy-windows:
    runs-on: windows-latest
    name: Deploy on windows
    needs: [test-windows]
    <<: *deploy_template