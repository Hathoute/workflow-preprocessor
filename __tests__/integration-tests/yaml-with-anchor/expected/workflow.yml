name: Workflow
'on':
  push:
    branches:
      - main
  pull_request:
    branches:
      - main
jobs:
  test-macos:
    runs-on: macos-latest
    name: Test on macos
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 16
      - name: Install dependencies
        run: npm install
      - name: Run tests
        run: npm test
  deploy-macos:
    runs-on: macos-latest
    name: Deploy on macos
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 16
      - name: Install dependencies
        run: npm install
      - name: Run deploy
        run: npm run deploy
    needs:
      - test-macos
  test-ubuntu:
    runs-on: ubuntu-latest
    name: Test on ubuntu
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 16
      - name: Install dependencies
        run: npm install
      - name: Run tests
        run: npm test
  deploy-ubuntu:
    runs-on: ubuntu-latest
    name: Deploy on ubuntu
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 16
      - name: Install dependencies
        run: npm install
      - name: Run deploy
        run: npm run deploy
    needs:
      - test-ubuntu
  test-windows:
    runs-on: windows-latest
    name: Test on windows
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 16
      - name: Install dependencies
        run: npm install
      - name: Run tests
        run: npm test
  deploy-windows:
    runs-on: windows-latest
    name: Deploy on windows
    steps:
      - name: Checkout
        uses: actions/checkout@v2
      - name: Setup Node.js
        uses: actions/setup-node@v2
        with:
          node-version: 16
      - name: Install dependencies
        run: npm install
      - name: Run deploy
        run: npm run deploy
    needs:
      - test-windows